src/controllers/Controller.js

```diff
@@ -0,0 +1,112 @@
+import { Console } from "@woowacourse/mission-utils";
+
+import Opponent from "../models/Opponent.js";
+import User from "../models/User.js";
+
+import GAME_START_VIEW from "../views/GameStartView.js";
+import INPUT_NUMBER_VIEW from "../views/InputNumberView.js";
+import OUTPUT_VIEW from "../views/OutputView.js";
+
+import RANDOM_NUMBER from "../utils/RandomNumber.js";
+import CHECK_VALIDATION from "../utils/CheckValidation.js";
+
+import MESSAGES from "../constants/Messages.js";
+import CONDITIONS from "../constants/Conditions.js";
+import RESTART_VIEW from "../views/RestartView.js";
+
+/**
+ * @class Controller
+ * 1. 상대방(컴퓨터) 숫자를 생성한다
+ * 2. 상대방의 숫자와 사용자가 입력한 숫자를 비교하여 평가한다
+ */
+class Controller {
+  /** @type {Opponent} */
+  #opponent;
+
+  /** @type {User} */
+  #user;
+
+  /**
+   * 게임을 실행하는 함수
+   * @returns {boolean} 재시작 여부
+   */
+  async run() {
+    // 게임 시작
+    GAME_START_VIEW();
+    this.generateOpponent();
+
+    // 숫자 입력 후 평가
+    this.#user = new User();
+    await this.matchNumber();
+
+    // 재시작 여부 입력
+    await this.saveUserRestart();
+    return this.#user.getRestart();
+  }
+
+  /**
+   * 상대방(컴퓨터) 숫자를 생성하는 함수
+   * @returns {Opponent}
+   */
+  generateOpponent() {
+    this.#opponent = new Opponent(RANDOM_NUMBER());
+    return this.#opponent;
+  }
+
+  /**
+   * 상대방(컴퓨터)의 숫자와 사용자의 숫자를 비교하여 평가하는 함수
+   */
+  async matchNumber() {
+    const opponentNumberElements = this.#opponent
+      .getNumber()
+      .toString()
+      .split("")
+      .map(Number);
+
+    let strikeCount = 0;
+    let ballCount = 0;
+    while (strikeCount !== CONDITIONS.MAX_STRIKE_COUNT) {
+      strikeCount = 0;
+      ballCount = 0;
+
+      const userNumber = await this.saveUserInput();
+      const userNumberElements = userNumber.toString().split("").map(Number);
```

Comment:
저는 이렇게 한번 작성해봤습니다!

```suggestion
      const userNumberElements = Array.from(String(userNumber),Number);
```

**[참고자료]**
https://github.com/parksb/javascript-style-guide#arrays--mapping 
https://github.com/parksb/javascript-style-guide#coercion--strings

Reply:
해당 코드로도 작성할 수 있겠네요!ㅎㅎ 감사합니다!

Reply:
utils 폴더에는 여러 class에서 재사용성이 높을 것 같은 함수들로 구성하고자 했는데, 다시 보니 RandomNumber 함수는 utils 폴더 보다는 Controller 클래스에 구현하는 것이 나았겠다는 생각이 드네요 ㅎㅎ 감사합니다:)